doc/api/http2.md

```diff
@@ -48,6 +48,7 @@ server.on('error', (err) => console.error(err));
 
 server.on('stream', (stream, headers) => {
   // stream is a Duplex
+  stream.resume();
```

Comment:
This would need a comment on why.

Reply:
As we said on #25952, this is not needed on server streams. So I'm reverting this change and adding a note in the stream lifecycle section.